lib/core/msdf-sys/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+    pub data   : [f32; Self::MAX_DATA_SIZE]
+}
+
+// Parameters of MSDF generation
+//
+// The structure gathering MSDF generation parameters meant to be same for all rendered glyphs
+pub struct MSDFParameters {
+    pub width                         : usize,
+    pub height                        : usize,
+    pub edge_coloring_angle_threshold : f64,
+    pub range                         : f64,
+    pub edge_threshold                : f64,
+    pub overlap_support               : bool
+}
+
+impl MutlichannelSignedDistanceField {
+    pub const MAX_SIZE       : usize = 32;
+    pub const CHANNELS_COUNT : usize = 3;
+    pub const MAX_DATA_SIZE  : usize = Self::MAX_SIZE * Self::MAX_SIZE *
+        Self::CHANNELS_COUNT;
+
+    pub fn generate(
+        font      : &Font,
+        unicode   : u32,
+        params    : MSDFParameters,
+        scale     : Vector2D<f64>,
+        translate : Vector2D<f64>,
+    ) -> MutlichannelSignedDistanceField {
+        assert!(params.width  <= Self::MAX_SIZE);
+        assert!(params.height <= Self::MAX_SIZE);
+
+        let output_address = msdfgen_generate_msdf(
+            params.width,
+            params.height,
+            font.handle.clone(),
+            unicode,
+            params.edge_coloring_angle_threshold,
+            params.range,
+            scale.x,
+            scale.y,
+            translate.x,
+            translate.y,
+            params.edge_threshold,
+            params.overlap_support
+        );
+        let mut data : [f32; Self::MAX_DATA_SIZE] = [0.0; Self::MAX_DATA_SIZE];
+        let data_size = params.width*params.height*Self::CHANNELS_COUNT;
+        copy_f32_data_from_msdfgen_memory(
+            output_address,
+            &mut data,
+            data_size
+        );
+        MutlichannelSignedDistanceField {
+            width: params.width,
+            height: params.height,
+            data
+        }
+    }
+}
+
+
+
+#[cfg(test)]
+mod tests {
+    extern crate wasm_bindgen_test;
+    extern crate slice_as_array;
+    use wasm_bindgen_test::wasm_bindgen_test;
+    use slice_as_array::slice_to_array_clone;
+    use internal::msdfgen_max_msdf_size;
+    use crate::*;
+
+    #[wasm_bindgen_test]
+    fn generate_msdf_for_capital_a() {
+        // given
+        let test_font : &[u8] = include_bytes!("DejaVuSansMono-Bold.ttf");
+        let expected_output_raw : &[u8] = include_bytes!("output.bin");
+        let font = Font::load_from_memory(test_font);
+        let params = MSDFParameters {
+            width: 32,
+            height: 32,
+            edge_coloring_angle_threshold: 3.0,
+            range: 2.0,
+            edge_threshold: 1.001,
+            overlap_support: true
+        };
+        // when
+        let msdf = MutlichannelSignedDistanceField::generate(
+            &font,
+            'A' as u32,
+            params,
+            Vector2D { x: 1.0, y: 1.0 },
+            Vector2D { x: 0.0, y: 0.0 }
+        );
+        // then
```

Comment:
What do these "when" and "get" comments mean?

Reply:
These are sections inspired by Spock:
given - a test setup and parameters
when - the thing we're testing
then - checking assertions